js/classes/entities/midground/npc/village/blacksmith.js

```diff
@@ -0,0 +1,140 @@
+
+                        // Is there overlap with the block on the x or y-axes?
+                        const isOverlapX = this.lastBoundingBox.left < entity.boundingBox.right
+                            && this.lastBoundingBox.right > entity.boundingBox.left;
+                        const isOverlapY = this.lastBoundingBox.bottom > entity.boundingBox.top
+                            && this.lastBoundingBox.top < entity.boundingBox.bottom;
+
+                        if (isOverlapX
+                            && this.lastBoundingBox.bottom <= entity.boundingBox.top
+                            && this.boundingBox.bottom > entity.boundingBox.top) {
+                            // We are colliding with the top.
+
+                            this.pos = new Vector(this.pos.x, entity.boundingBox.top - BlackSmith.SCALED_SIZE.y);
+                            this.velocity = new Vector(this.velocity.x, 0);
+                        } else if (isOverlapY
+                            && this.lastBoundingBox.right <= entity.boundingBox.left
+                            && this.boundingBox.right > entity.boundingBox.left) {
+                            // We are colliding with the left side.
+
+                            this.pos = new Vector(entity.boundingBox.left - BlackSmith.SCALED_SIZE.x, this.pos.y);
+                        } else if (isOverlapY
+                            && this.lastBoundingBox.left >= entity.boundingBox.right
+                            && this.boundingBox.left < entity.boundingBox.right) {
+                            // We are colliding with the right side.
+
+                            this.pos = new Vector(entity.boundingBox.right, this.pos.y);
+                        } else if (isOverlapX
+                            && this.lastBoundingBox.top >= entity.boundingBox.bottom
+                            && this.boundingBox.top < entity.boundingBox.bottom) {
+                            // We are colliding with the bottom.
+                            this.pos = new Vector(this.pos.x, entity.boundingBox.bottom);
+                        }
+                    }
+                }
+                // There's no collision - don't do anything!
+            }
+            // There's no bounding box, so who gives a shrek?
+        });
+        // Step 5: Now that your position is actually figured out, draw your correct bounding box.
+        this.boundingBox = new BoundingBox(this.pos, BlackSmith.SCALED_SIZE);
+    };
+
+    /** Draw Papa Chad on the canvas. */
+    draw() {
+        this.animations[this.facing][this.action].drawFrame(Vector.worldToCanvasSpace(this.pos), BlackSmith.SCALE);
+        if (this.conversation.new) {
+            const indicator = new OverheadIcon(this, BlackSmith.SCALED_SIZE.x, OverheadIcon.TRIANGLE, OverheadIcon.GREEN);
+            indicator.draw();
+        }
+    };
+
+    /** Called by the constructor. Fills up the animations array. */
+    loadAnimations() {
```

Comment:
You could get rid of this method by just creating a single instance field in the constructor (something like `this.animator`) and assigning it the animator you define at line 134.